forge-game/src/main/java/forge/game/card/Card.java

```diff
@@ -7073,14 +7073,8 @@ public final boolean canBeSacrificedBy(final SpellAbility source, final boolean
     }
 
     public final boolean canExiledBy(final SpellAbility source, final boolean effect) {
-        final Card gameCard = game.getCardState(this, null);
-        // gameCard is LKI in that case, the card is not in game anymore
-        // or the timestamp did change
-        // this should check Self too
-        if (gameCard == null || !this.equalsWithTimestamp(gameCard)) {
-            return false;
-        }
-
+        if (game.getCardState(this, null) == null) return false;
```

Comment:
Imo for the reasons I outlined in the issue removing this check completely is the cleaner option